src/kademlia/dht_tracker.cpp

```diff
@@ -242,15 +242,21 @@ namespace libtorrent { namespace dht {
 			add_dht_counters(n.second.dht, c);
 	}
 
-	void dht_tracker::connection_timeout(tracker_node& n, error_code const& e)
+	void dht_tracker::connection_timeout(aux::session_listen_socket* s, error_code const& e)
 	{
 		if (e || !m_running) return;
 
-		time_duration d = n.dht.connection_timeout();
+		auto const it = m_nodes.find(s);
+		// this could happen if the task is about to be executed (and not cancellable) and
+		// the socket is just removed
+		if (it == m_nodes.end()) return; // node already destroyed
```

Comment:
@arvidn I added the simulation test, this line is the critical one, if you comment it, the test will fail